main.go

```diff
@@ -245,12 +245,26 @@ func main() {
 
 	log.Info("Starting the controller managers")
 
-	mgrCtx := ctrl.SetupSignalHandler()
+	mainCtx := ctrl.SetupSignalHandler()
 	var wg sync.WaitGroup
 
+	webCtx, webCtxCancel := context.WithCancel(mainCtx)
+	mgrCtx, mgrCtxCancel := context.WithCancel(mainCtx)
+	hubMgrCtx, hubMgrCtxCancel := context.WithCancel(mainCtx)
```

Comment:
This seems like it works. I was expecting just one cancellable context that all 3 use, and an error in any of them cancels the context. Does that not work?

For reference, https://pkg.go.dev/context#pkg-overview
> The same Context may be passed to functions running in different goroutines; Contexts are safe for simultaneous use by multiple goroutines.

Reply:
I went back and forth on it but then thought it'd be easier to understand with a context per Go routine, but I don't have a strong opinion so I changed it to a single child context.